frontend/javascripts/oxalis/model/sagas/isosurface_saga.ts

```diff
@@ -962,10 +964,53 @@ function* downloadIsosurfaceCellById(
   }
 }
 
+function* downloadIsosurfaceCellsAsZIP(
+  arrayOfCells: { cellName: string; segmentId: number; layerName: string }[],
+): Saga<void> {
+  const { segmentMeshController } = getSceneController();
+  const zipWriter = new ZipWriter(new BlobWriter("application/zip"));
+  try {
+    const promises = arrayOfCells.map((element) => {
+      const geometry = segmentMeshController.getIsosurfaceGeometryInBestLOD(
+        element.segmentId,
+        element.layerName,
+      );
+
+      if (geometry == null) {
+        const errorMessage = messages["tracing.not_isosurface_available_to_download"];
+        Toast.error(errorMessage, {
+          sticky: false,
+        });
+        return;
+      }
+      const stlDataViews = exportToStl(geometry);
+      // Encode isosurface and cell id property
+      const { isosurfaceMarker, segmentIdIndex } = stlIsosurfaceConstants;
+      isosurfaceMarker.forEach((marker, index) => {
+        stlDataViews[0].setUint8(index, marker);
+      });
+      stlDataViews[0].setUint32(segmentIdIndex, element.segmentId, true);
+      const stlDataReader = new BlobReader(new Blob(stlDataViews));
```

Comment:
This is copied from somewhere, right? Can you extract the common code into a helper function?